src/realm/object-store/schema.cpp

```diff
@@ -58,11 +58,11 @@ Schema::Schema(base types) noexcept
 
 Schema::iterator Schema::find(StringData name) noexcept
 {
-    auto it = std::lower_bound(begin(), end(), name, [](ObjectSchema const& lft, StringData rgt) {
-        return lft.name < rgt;
-    });
-    if (it != end() && it->name != name) {
-        it = end();
+    auto it = find(*this, name);
+    if (it == end()) {
+        it = find(m_others, name);
```

Comment:
This does not work because `m_others` is not sorted.

Reply:
Why shouldn't `m_others` be sorted? Since `zip_matching` is basically a lexicographically merge function? I should probably add some test around this, probably I am missing something.